### ISSUE TITLE
Get the future forecast value.

### ISSUE THREAD

**Issue by wjmbalala** [ki-ljl/LSTM-Load-Forecasting#5]
Hello, good work. I have some doubts. The code is a prediction of testdata, and it doesn't output future results. How can I get the values that didn't happen in the future?

**Comment by ki-ljl** [comment]
After training, we can get an LSTM model, which we will call F. Today's date is May 7th. Suppose you use the previous seven days to predict the next day. Then you only need to construct a seq based on the load data from May 1st to May 7th and the environmental variable factors according to the method of constructing the training set. (without label), then use F(seq) to get the predicted value for May 8th.

**Comment by wjmbalala** [comment]
Thank you very much for your reply. I seem to understand it, but there is still a question. Are the data from No.1 to No.7 trained in the model?

**Comment by ki-ljl** [comment]
> 

Supervised training requires labels, this set of data has no labels, they are not trained.

**Comment by wjmbalala** [comment]
Thank you very much for your reply. I made the following changes to the code, and input the last 24 data of the data into the model to get four predicted values, but the last predicted dimension is 24*4. I don't know why, and I hope to get an answer.

def nn_seq(file_name, B, num):
    print('data processing...')
    data = load_data(file_name)
    load = data[data.columns[1]]
    load = load.tolist()
    data = data.values.tolist()
    seq = []
    seq_pred = []
    for i in range(0, len(data) - 24 - num, num):
        train_seq = []
        train_label = []
        for j in range(i, i + 24):
            x = [load[j]]
            for c in range(2, 8):
                x.append(data[j][c])
            train_seq.append(x)
        for j in range(i + 24, i + 24 + num):
            train_label.append(load[j])
        train_seq = torch.FloatTensor(train_seq)
        train_label = torch.FloatTensor(train_label).view(-1)
        seq.append((train_seq, train_label))
        # TODO
        seq_pred.append(train_seq)

    # print(seq[-1])

    Dtr = seq[0:int(len(seq) * 0.7)]
    Dte = seq[int(len(seq) * 0.7):len(seq)]
    # No label
    Dpred = seq_pred[int(len(seq_pred)-24):len(seq_pred)]

    train_len = int(len(Dtr) / B) * B
    test_len = int(len(Dte) / B) * B
    Dtr, Dte = Dtr[:train_len], Dte[:test_len]

    train = MyDataset(Dtr)
    test = MyDataset(Dte)
    # pred
    prediction = MyDataset(Dpred)

    Dtr = DataLoader(dataset=train, batch_size=B, shuffle=False, num_workers=0)
    Dte = DataLoader(dataset=test, batch_size=B, shuffle=False, num_workers=0)
    Dpred = DataLoader(dataset=prediction, batch_size=1, shuffle=False, num_workers=0)

    return Dtr, Dte, Dpred

# TODO prediction
def prediction(name, num):
    global MAX, MIN
    Dtr, Dte, Dpred = nn_seq(file_name=name, B=1, num=num)
    pred = []
    print('loading model...')
    input_size, hidden_size, num_layers = 7, 64, 1
    output_size = num
    model = LSTM(input_size, hidden_size, num_layers, output_size, batch_size=1).to(device)
    model.load_state_dict(torch.load(LSTM_PATH)['model'])
    model.eval()
    print('predicting...')
    for seq in Dpred:
        seq = seq.to(device)
        with torch.no_grad():
            y_pred = model(seq)
            print(y_pred)
            y_pred = list(chain.from_iterable(y_pred.data.tolist()))
            pred.extend(y_pred)

    pred = np.array([pred])
    pred = (MAX - MIN) * pred + MIN

    preds = pred.squeeze(0)
    save_txt('data/prediction.txt', preds)
![1](https://user-images.githubusercontent.com/61644613/167243266-6d455e6e-96e6-4c73-a612-5295122f7326.png)
